FairModule::ConstructASCIIGeometry: "warning: parameter 'dataType1' set but not used"

```console
base/sim/FairModule.h:170:44: warning: parameter 'dataType1' set but not used [-Wunused-but-set-parameter]
 void FairModule::ConstructASCIIGeometry(T* dataType1, TString containerName, U*)
                                         ~~~^~~~~~~~~
```

* FairRoot dev branch
* Added `-Wunused-but-set-parameter` in `cmake/modules/CheckCompiler.cmake`
* FairSoft's `test/env/fairroot_develop`
* Debian-10

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -140,6 +140,8 @@ file an issue, so that we can see how to handle this.
   * `FairFieldFactory::fCreator` points to `this`.
   * `FairRootManager::FillEventHeader` is only a wrapper around
     `FairSource::FillEventHeader`.
+  * `FairModule::ConstructASCIIGeometry(T*, TString, U*)`, use
+    `FairModule::ConstructASCIIGeometry(TString)` instead.
 * Many items were already deprecated in prior versions.
   Marked them with proper C++14 deprecation warnings.
   Scheduled them for removal in v20.

--- a/base/sim/FairModule.h
+++ b/base/sim/FairModule.h
@@ -1,5 +1,5 @@
 /********************************************************************************
- *    Copyright (C) 2014 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH    *
+ * Copyright (C) 2014-2022 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH  *
  *                                                                              *
  *              This software is distributed under the terms of the             *
  *              GNU Lesser General Public Licence (LGPL) version 3,             *
@@ -90,9 +90,14 @@ class FairModule : public TNamed
     /** Finish worker run (used in MT mode only) */
     virtual void FinishWorkerRun() const { ; }
 
-    /**template function to construct geometry. to be used in derived classes.*/
+    /** @deprecated template function to construct geometry. to be used in derived classes.
+     * The first and the third argument are meaningless, just pass nullptr */
     template<class T, class U>
-    void ConstructASCIIGeometry(T* dataType1, TString containerName = "", U* datatype2 = nullptr);
+    [[deprecated("Broken signature, use ConstructASCIIGeometry(TString) instead")]] void
+        ConstructASCIIGeometry(T*, TString containerName = "", U* = nullptr);
+    /** Helper function to construct geometry. */
+    template<class T, class U>
+    void ConstructASCIIGeometry(TString containerName = "");
 
     /**Set the sensitivity flag for volumes, called from ConstructASCIIRootGeometry(), and has to be implimented for
      * detectors which use ConstructASCIIRootGeometry() to build the geometry */
@@ -169,14 +174,14 @@ class FairModule : public TNamed
 };
 
 template<class T, class U>
-void FairModule::ConstructASCIIGeometry(T* dataType1, TString containerName, U*)
+void FairModule::ConstructASCIIGeometry(TString containerName)
 {
     FairGeoLoader* loader = FairGeoLoader::Instance();
     FairGeoInterface* GeoInterface = loader->getGeoInterface();
     T* MGeo = new T();
     MGeo->print();
     MGeo->setGeomFile(GetGeometryFileName());
-    GeoInterface->addGeoModule(MGeo);
+    GeoInterface->addGeoModule(MGeo);   // takes ownership!
     Bool_t rc = GeoInterface->readSet(MGeo);
     if (rc) {
         MGeo->create(loader->getGeoBuilder());
@@ -186,8 +191,6 @@ void FairModule::ConstructASCIIGeometry(T* dataType1, TString containerName, U*)
     // store geo parameter
     FairRun* fRun = FairRun::Instance();
     FairRuntimeDb* rtdb = FairRun::Instance()->GetRuntimeDb();
-
-    dataType1 = MGeo;
 
     if ("" != containerName) {
         LOG(info) << "Add GeoNodes for " << MGeo->getDescription() << " to container " << containerName;
@@ -213,6 +216,12 @@ void FairModule::ConstructASCIIGeometry(T* dataType1, TString containerName, U*)
         par->setChanged();
         par->setInputVersion(fRun->GetRunId(), 1);
     }
+}
+
+template<class T, class U>
+void FairModule::ConstructASCIIGeometry(T*, TString containerName, U*)
+{
+    ConstructASCIIGeometry<T, U>(containerName);
 }
 
 #endif   // FAIRMODULE_H

--- a/examples/MQ/pixelDetector/src/Pixel.cxx
+++ b/examples/MQ/pixelDetector/src/Pixel.cxx
@@ -1,5 +1,5 @@
 /********************************************************************************
- *    Copyright (C) 2014 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH    *
+ * Copyright (C) 2014-2022 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH  *
  *                                                                              *
  *              This software is distributed under the terms of the             *
  *              GNU Lesser General Public Licence (LGPL) version 3,             *
@@ -160,8 +160,7 @@ void Pixel::ConstructGeometry()
       just copy this and use it for your detector, otherwise you can
       implement here you own way of constructing the geometry. */
 
-    PixelGeo* Geo = new PixelGeo();
-    ConstructASCIIGeometry<PixelGeo, PixelGeoPar>(Geo, "PixelGeoPar");
+    ConstructASCIIGeometry<PixelGeo, PixelGeoPar>("PixelGeoPar");
 }
 
 Bool_t Pixel::IsSensitive(const std::string& name)

--- a/examples/advanced/Tutorial3/simulation/FairTestDetector.cxx
+++ b/examples/advanced/Tutorial3/simulation/FairTestDetector.cxx
@@ -1,5 +1,5 @@
 /********************************************************************************
- *    Copyright (C) 2014 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH    *
+ * Copyright (C) 2014-2022 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH  *
  *                                                                              *
  *              This software is distributed under the terms of the             *
  *              GNU Lesser General Public Licence (LGPL) version 3,             *
@@ -148,8 +148,7 @@ void FairTestDetector::ConstructGeometry()
         just copy this and use it for your detector, otherwise you can
         implement here you own way of constructing the geometry. */
 
-    FairTestDetectorGeo* Geo = new FairTestDetectorGeo();
-    ConstructASCIIGeometry<FairTestDetectorGeo, FairTestDetectorGeoPar>(Geo, "FairTestDetectorGeoPar");
+    ConstructASCIIGeometry<FairTestDetectorGeo, FairTestDetectorGeoPar>("FairTestDetectorGeoPar");
 }
 
 FairTestDetectorPoint* FairTestDetector::AddHit(Int_t trackID,

--- a/examples/advanced/propagator/src/FairTutPropDet.cxx
+++ b/examples/advanced/propagator/src/FairTutPropDet.cxx
@@ -1,5 +1,5 @@
 /********************************************************************************
- *    Copyright (C) 2019 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH    *
+ * Copyright (C) 2019-2022 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH  *
  *                                                                              *
  *              This software is distributed under the terms of the             *
  *              GNU Lesser General Public Licence (LGPL) version 3,             *
@@ -146,8 +146,7 @@ void FairTutPropDet::ConstructGeometry()
         just copy this and use it for your detector, otherwise you can
         implement here you own way of constructing the geometry. */
 
-    FairTutPropGeo* Geo = new FairTutPropGeo();
-    ConstructASCIIGeometry<FairTutPropGeo, FairTutPropGeoPar>(Geo, "FairTutPropGeoPar");
+    ConstructASCIIGeometry<FairTutPropGeo, FairTutPropGeoPar>("FairTutPropGeoPar");
 }
 
 FairTutPropPoint* FairTutPropDet::AddHit(Int_t trackID,

--- a/examples/common/passive/FairMagnet.cxx
+++ b/examples/common/passive/FairMagnet.cxx
@@ -1,5 +1,5 @@
 /********************************************************************************
- *    Copyright (C) 2014 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH    *
+ * Copyright (C) 2014-2022 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH  *
  *                                                                              *
  *              This software is distributed under the terms of the             *
  *              GNU Lesser General Public Licence (LGPL) version 3,             *
@@ -49,9 +49,7 @@ Bool_t FairMagnet::IsSensitive(const std::string& /*name*/)
 
 void FairMagnet::ConstructASCIIGeometry()
 {
-    FairGeoMagnet* MGeo = new FairGeoMagnet();
-
-    FairModule::ConstructASCIIGeometry<FairGeoMagnet, FairGeoPassivePar>(MGeo, "FairGeoPassivePar");
+    FairModule::ConstructASCIIGeometry<FairGeoMagnet, FairGeoPassivePar>("FairGeoPassivePar");
 }
 
 FairModule* FairMagnet::CloneModule() const { return new FairMagnet(*this); }

--- a/examples/common/passive/FairTarget.cxx
+++ b/examples/common/passive/FairTarget.cxx
@@ -1,5 +1,5 @@
 /********************************************************************************
- *    Copyright (C) 2014 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH    *
+ * Copyright (C) 2014-2022 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH  *
  *                                                                              *
  *              This software is distributed under the terms of the             *
  *              GNU Lesser General Public Licence (LGPL) version 3,             *
@@ -27,9 +27,7 @@ FairTarget::FairTarget(const FairTarget& rhs)
 
 void FairTarget::ConstructGeometry()
 {
-    FairGeoTarget* MGeo = new FairGeoTarget();
-
-    ConstructASCIIGeometry<FairGeoTarget, FairGeoPassivePar>(MGeo, "FairGeoPassivePar");
+    ConstructASCIIGeometry<FairGeoTarget, FairGeoPassivePar>("FairGeoPassivePar");
 }
 
 FairModule* FairTarget::CloneModule() const { return new FairTarget(*this); }

--- a/examples/simulation/Tutorial1/src/FairTutorialDet1.cxx
+++ b/examples/simulation/Tutorial1/src/FairTutorialDet1.cxx
@@ -20,8 +20,6 @@
 #include <TVirtualMC.h>        // for TVirtualMC
 #include <TVirtualMCStack.h>   // for TVirtualMCStack
 
-FairTutorialDet1Geo* FairTutorialDet1::fgGeo = nullptr;
-
 FairTutorialDet1::FairTutorialDet1()
     : FairDetector("TutorialDet", kTRUE, kTutDet)
 {}
@@ -129,7 +127,7 @@ void FairTutorialDet1::ConstructGeometry()
       just copy this and use it for your detector, otherwise you can
       implement here you own way of constructing the geometry. */
 
-    ConstructASCIIGeometry<FairTutorialDet1Geo, FairTutorialDet1GeoPar>(fgGeo, "FairTutorialDet1GeoPar");
+    ConstructASCIIGeometry<FairTutorialDet1Geo, FairTutorialDet1GeoPar>("FairTutorialDet1GeoPar");
 }
 
 FairTutorialDet1Point* FairTutorialDet1::AddHit(Int_t trackID,

--- a/examples/simulation/Tutorial1/src/FairTutorialDet1.h
+++ b/examples/simulation/Tutorial1/src/FairTutorialDet1.h
@@ -82,8 +82,6 @@ class FairTutorialDet1 : public FairDetector
     Bool_t IsSensitive(const std::string& name) override;
 
   private:
-    static FairTutorialDet1Geo* fgGeo;   //!
-
     /** Track information to be stored until the track leaves the
     active volume.
     */

--- a/examples/simulation/Tutorial2/src/FairTutorialDet2.cxx
+++ b/examples/simulation/Tutorial2/src/FairTutorialDet2.cxx
@@ -1,5 +1,5 @@
 /********************************************************************************
- *    Copyright (C) 2014 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH    *
+ * Copyright (C) 2014-2022 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH  *
  *                                                                              *
  *              This software is distributed under the terms of the             *
  *              GNU Lesser General Public Licence (LGPL) version 3,             *
@@ -22,8 +22,6 @@
 #ifdef NDEBUG
 #undef NDEBUG
 #endif
-
-FairTutorialDet2Geo* FairTutorialDet2::fgGeo;   //!
 
 FairTutorialDet2::FairTutorialDet2()
     : FairDetector("TutorialDet", kTRUE, kTutDet)
@@ -159,7 +157,7 @@ void FairTutorialDet2::ConstructGeometry()
       just copy this and use it for your detector, otherwise you can
       implement here you own way of constructing the geometry. */
 
-    ConstructASCIIGeometry<FairTutorialDet2Geo, FairTutorialDet2GeoPar>(fgGeo, "FairTutorialDet2GeoPar");
+    ConstructASCIIGeometry<FairTutorialDet2Geo, FairTutorialDet2GeoPar>("FairTutorialDet2GeoPar");
 }
 
 FairTutorialDet2Point* FairTutorialDet2::AddHit(Int_t trackID,

--- a/examples/simulation/Tutorial2/src/FairTutorialDet2.h
+++ b/examples/simulation/Tutorial2/src/FairTutorialDet2.h
@@ -1,5 +1,5 @@
 /********************************************************************************
- *    Copyright (C) 2014 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH    *
+ * Copyright (C) 2014-2022 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH  *
  *                                                                              *
  *              This software is distributed under the terms of the             *
  *              GNU Lesser General Public Licence (LGPL) version 3,             *
@@ -80,8 +80,6 @@ class FairTutorialDet2 : public FairDetector
     virtual FairModule* CloneModule() const;
 
   private:
-    static FairTutorialDet2Geo* fgGeo;   //!
-
     /** Track information to be stored until the track leaves the
     active volume.
     */

--- a/examples/simulation/Tutorial4/src/mc/FairTutorialDet4.cxx
+++ b/examples/simulation/Tutorial4/src/mc/FairTutorialDet4.cxx
@@ -1,5 +1,5 @@
 /********************************************************************************
- *    Copyright (C) 2014 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH    *
+ * Copyright (C) 2014-2022 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH  *
  *                                                                              *
  *              This software is distributed under the terms of the             *
  *              GNU Lesser General Public Licence (LGPL) version 3,             *
@@ -24,8 +24,6 @@
 #include <TString.h>           // for TString, operator<<, Form
 #include <TVirtualMC.h>        // for TVirtualMC
 #include <TVirtualMCStack.h>   // for TVirtualMCStack
-
-FairTutorialDet4Geo* FairTutorialDet4::fgGeo;   //!
 
 FairTutorialDet4::FairTutorialDet4()
     : FairDetector("TutorialDet", kTRUE, kTutDet)
@@ -261,7 +259,7 @@ void FairTutorialDet4::ConstructASCIIGeometry()
       just copy this and use it for your detector, otherwise you can
       implement here you own way of constructing the geometry. */
 
-    FairModule::ConstructASCIIGeometry<FairTutorialDet4Geo, FairTutorialDet4GeoPar>(fgGeo, "FairTutorialDet4GeoPar");
+    FairModule::ConstructASCIIGeometry<FairTutorialDet4Geo, FairTutorialDet4GeoPar>("FairTutorialDet4GeoPar");
 }
 
 std::map<std::string, TGeoHMatrix> FairTutorialDet4::getMisalignmentMatrices()

--- a/examples/simulation/Tutorial4/src/mc/FairTutorialDet4.h
+++ b/examples/simulation/Tutorial4/src/mc/FairTutorialDet4.h
@@ -1,5 +1,5 @@
 /********************************************************************************
- *    Copyright (C) 2014 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH    *
+ * Copyright (C) 2014-2022 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH  *
  *                                                                              *
  *              This software is distributed under the terms of the             *
  *              GNU Lesser General Public Licence (LGPL) version 3,             *
@@ -98,8 +98,6 @@ class FairTutorialDet4 : public FairDetector
     virtual Bool_t IsSensitive(const std::string& name);
 
   private:
-    static FairTutorialDet4Geo* fgGeo;   //!
-
     /** Track information to be stored until the track leaves the
     active volume.
     */

--- a/examples/simulation/rutherford/src/FairRutherford.cxx
+++ b/examples/simulation/rutherford/src/FairRutherford.cxx
@@ -1,5 +1,5 @@
 /********************************************************************************
- *    Copyright (C) 2014 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH    *
+ * Copyright (C) 2014-2022 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH  *
  *                                                                              *
  *              This software is distributed under the terms of the             *
  *              GNU Lesser General Public Licence (LGPL) version 3,             *
@@ -18,8 +18,6 @@
 #include <TClonesArray.h>      // for TClonesArray
 #include <TVirtualMC.h>        // for TVirtualMC
 #include <TVirtualMCStack.h>   // for TVirtualMCStack
-
-FairRutherfordGeo* FairRutherford::fgGeo = nullptr;
 
 FairRutherford::FairRutherford()
     : FairDetector("FairRutherford", kTRUE, kFairRutherford)
@@ -124,7 +122,7 @@ void FairRutherford::ConstructGeometry()
       just copy this and use it for your detector, otherwise you can
       implement here you own way of constructing the geometry. */
 
-    ConstructASCIIGeometry<FairRutherfordGeo, FairRutherfordGeoPar>(fgGeo, "FairRutherfordGeoPar");
+    ConstructASCIIGeometry<FairRutherfordGeo, FairRutherfordGeoPar>("FairRutherfordGeoPar");
 }
 
 FairRutherfordPoint* FairRutherford::AddHit(Int_t trackID,

--- a/examples/simulation/rutherford/src/FairRutherford.h
+++ b/examples/simulation/rutherford/src/FairRutherford.h
@@ -1,5 +1,5 @@
 /********************************************************************************
- *    Copyright (C) 2014 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH    *
+ * Copyright (C) 2014-2022 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH  *
  *                                                                              *
  *              This software is distributed under the terms of the             *
  *              GNU Lesser General Public Licence (LGPL) version 3,             *
@@ -79,8 +79,6 @@ class FairRutherford : public FairDetector
     FairModule* CloneModule() const override;
 
   private:
-    static FairRutherfordGeo* fgGeo;   //!
-
     /** Track information to be stored until the track leaves the
     active volume.
     */

